### PR TITLE
8303948: HsErrFileUtils.checkHsErrFileContent() fails to check the last pattern.

### PR DESCRIPTION
**Pull request by varada1110**
When last pattern in deque [positivePatternStack] is not matching in HsErrFile, it comes out of the loop and check whether the positivePatternStack is empty or not, which turns to be empty because the pollFirst() removes the pattern. 

This has been noticed in the TestSigInfoInHsErrFile.java where the segfault address for AIX is set as "0x0*1400" instead of "0xffffffffffffffff", which should throw the expected error but the error is neglected due to empty deque and the test is passed. 

JBS issue : [8303948](https://bugs.openjdk.org/browse/JDK-8303948)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8303948](https://bugs.openjdk.org/browse/JDK-8303948): HsErrFileUtils.checkHsErrFileContent() fails to check the last pattern.

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to [871b2e42](https://git.openjdk.org/jdk/pull/12970/files/871b2e42116c5f2ce2e3b9420c88e82b71752c1d)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/12970/head:pull/12970` \
`$ git checkout pull/12970`

Update a local copy of the PR: \
`$ git checkout pull/12970` \
`$ git pull https://git.openjdk.org/jdk.git pull/12970/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12970`

View PR using the GUI difftool: \
`$ git pr show -t 12970`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12970.diff">https://git.openjdk.org/jdk/pull/12970.diff</a>

</details>

### CODE REVIEW

**Review by dholmes-ora — APPROVED** [review]
This fix seems correct - thanks.

**Comment by varada1110** [comment]
> This fix seems correct - thanks.

Thanks @dholmes-ora for reviewing the changes.

**Comment by dholmes-ora** [comment]
Hotspot changes, including tests, require two reviewers. I'll try to get someone else to review and sponsor once done.

**Review by tstuefe — APPROVED** [review]
@varada1110 Ouch, thank you for finding that. Let's see how many tests now start failing :-/

Could a slightly more elegant patch not be to change the condition after the loop not from
```
< if (!positivePatternStack.isEmpty()) {
> if (currentPositivePattern != null) {
```
?

But I leave that up to you. Patch is also fine in its current form.

**Comment by varada1110** [comment]
> @varada1110 Ouch, thank you for finding that. Let's see how many tests now start failing :-/
> 
> Could a slightly more elegant patch not be to change the condition after the loop not from
> 
> ```
> < if (!positivePatternStack.isEmpty()) {
> > if (currentPositivePattern != null) {
> ```
> 
> ?
> 
> But I leave that up to you. Patch is also fine in its current form.

Thank you @tstuefe for the suggestion. I have made the changes please review it

**Review by tstuefe — APPROVED** [review]
Still good.

File: test/hotspot/jtreg/runtime/ErrorHandling/HsErrFileUtils.java [hunk 1]
```diff
@@ -131,8 +130,8 @@ public static void checkHsErrFileContent(File f, Pattern[] positivePatterns, Pat
                 lastLine = line;
                 lineNo++;
             }
-            // Found all positive pattern?
-            if (!positivePatternStack.isEmpty()) {
+            // check if current pattern is NULL then pattern matches else throw RuntimeException
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/runtime/ErrorHandling/HsErrFileUtils.java:133]: Sorry this doesn't read very clearly to me. Suggestion:
```
// If the current pattern is not null then it didn't match
```
Thanks

**Review by dholmes-ora — APPROVED** [review]
So checking for null-ness covers the case of the last pattern not matching, regardless of whether there are any other patterns still to be checked. Yep that logic seems fine.

A nit with the explanatory comment though.

File: test/hotspot/jtreg/runtime/ErrorHandling/HsErrFileUtils.java (hunk 1 above)
Comment (varada1110) [reply test/hotspot/jtreg/runtime/ErrorHandling/HsErrFileUtils.java]: Thank you @dholmes-ora for the suggestion. I have updated the comment. Please review the change.

**Comment by tstuefe** [comment]
Still good to go

**Review by dholmes-ora — APPROVED** [review]
Looks good. Thanks. I will sponsor now.

**Comment by varada1110** [comment]
> @dholmes-ora The PR has been updated since the change author (@varada1110) issued the `integrate` command - the author must perform this command again.

/integrate

**Pull request closed without merging**